models/src/node/client.rs

```diff
@@ -0,0 +1,234 @@
+    #[serde(rename = "apiVersion")]
+    api_version: String,
+    kind: String,
+    status: BottlerocketNodeStatus,
+}
+
+impl Default for BottlerocketNodeStatusPatch {
+    fn default() -> Self {
+        BottlerocketNodeStatusPatch {
+            api_version: constants::API_VERSION.to_string(),
+            kind: K8S_NODE_KIND.to_string(),
+            status: BottlerocketNodeStatus::default(),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+/// A helper struct used to serialize and send patches to the k8s API to modify the spec of a BottlerocketNode.
+struct BottlerocketNodeSpecPatch {
+    #[serde(rename = "apiVersion")]
+    api_version: String,
+    kind: String,
+    spec: BottlerocketNodeSpec,
+}
+
+impl Default for BottlerocketNodeSpecPatch {
+    fn default() -> Self {
+        BottlerocketNodeSpecPatch {
+            api_version: constants::API_VERSION.to_string(),
+            kind: K8S_NODE_KIND.to_string(),
+            spec: BottlerocketNodeSpec::default(),
+        }
+    }
+}
+
+#[async_trait]
+impl BottlerocketNodeClient for K8SBottlerocketNodeClient {
+    #[instrument(skip(self), err)]
+    async fn create_node(&self, selector: &BottlerocketNodeSelector) -> Result<BottlerocketNode> {
+        let br_node = BottlerocketNode {
+            metadata: ObjectMeta {
+                name: Some(selector.brn_resource_name()),
+                owner_references: Some(vec![OwnerReference {
+                    api_version: "v1".to_string(),
+                    kind: "Node".to_string(),
+                    name: selector.node_name.clone(),
+                    uid: selector.node_uid.clone(),
+                    ..Default::default()
+                }]),
+                ..Default::default()
+            },
+            spec: BottlerocketNodeSpec::default(),
+            ..Default::default()
+        };
+
+        Api::namespaced(self.k8s_client.clone(), constants::NAMESPACE)
+            .create(&PostParams::default(), &br_node)
+            .await
+            .map_err(|err| Box::new(err) as Box<dyn std::error::Error>)
+            .context(error::CreateBottlerocketNode {
+                selector: selector.clone(),
+            })?;
+
+        Ok(br_node)
+    }
+
+    #[instrument(skip(self), err)]
+    async fn update_node_status(
+        &self,
+        selector: &BottlerocketNodeSelector,
+        status: &BottlerocketNodeStatus,
+    ) -> Result<()> {
+        let br_node_status_patch = BottlerocketNodeStatusPatch {
+            status: status.clone(),
+            ..Default::default()
+        };
+
+        let br_node_status_patch =
+            serde_json::to_value(br_node_status_patch).context(error::CreateK8SPatch)?;
+
+        let api: Api<BottlerocketNode> =
+            Api::namespaced(self.k8s_client.clone(), constants::NAMESPACE);
+
+        api.patch_status(
+            &selector.brn_resource_name(),
+            &PatchParams::default(),
+            &Patch::Merge(&br_node_status_patch),
```

Comment:
Do you need to transform `br_node_status_patch`? It looks like `Patch::Merge()` receives an object that derives from `Serialize`, and `BottlerocketNodeStatusPatch` already derives from it.

Reply:
Nice catch! Fixed.